phpBB/includes/acp/acp_main.php

```diff
@@ -679,17 +679,17 @@ function main($id, $mode)
 			}
 		}
 
-		if (!defined('PHPBB_DISABLE_CONFIG_CHECK') && file_exists($phpbb_root_path . 'config.' . $phpEx) && $phpbb_filesystem->is_writable($phpbb_root_path . 'config.' . $phpEx))
+		if (!defined('PHPBB_DISABLE_CONFIG_CHECK'))
 		{
-			// World-Writable? (000x)
-			$template->assign_var('S_WRITABLE_CONFIG', (bool) (@fileperms($phpbb_root_path . 'config.' . $phpEx) & 0x0002));
+			$template->assign_var('S_WRITABLE_CONFIG', (bool) $phpbb_filesystem->is_writable($phpbb_root_path . 'config.' . $phpEx));
```

Comment:
The previous code checked whether the config file is world writable while your new code checks whether it's writable which would also apply when it's not actually world writable?

Reply:
Now it deals with Home PC servers, FTP etc., we had this doubt in effect.

Reply:
Now it actually still says config.php is world writable when it isn't. ;)

Reply:
The issue is with the switch `S_WRITABLE_CONFIG` and Home servers on windows, I think the actual should code cover all cases. Home servers on windows can not use the FTP facilities (chmod) but rely on the system ones (read-only).

Are you saying you tested it on linux and it doesn't works as expected?